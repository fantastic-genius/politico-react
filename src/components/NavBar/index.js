import React from 'react';
import  { Link } from 'react-router-dom';
import Button from '@components/Button';

const NavBar = () => (
  <div>
    <header>
        <div className="app-name">
          <Link to='/'>
            <h2>POLITICO</h2>
          </Link>
        </div>
        <div className="sign-btn">
            <span className="home-btn">
              <Link to='/signup'>
                <Button className="btn" value='Sign Up'/>
              </Link>
              <Link to='login'>
                <Button className="btn" value='Login'/>
              </Link>
            </span>
        </div>
    </header>
  </div>
);

export default NavBar;
