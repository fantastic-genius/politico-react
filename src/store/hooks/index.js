import { useEffect } from 'react';
import  { setUser } from '@actions/Auth';

const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(setUser());
  }, []);
};

export default useSetUser;