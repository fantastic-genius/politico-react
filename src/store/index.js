import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger as logger } from 'redux-logger';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reducer from './reducers';

const middleware = applyMiddleware(promise, thunk, logger());
const store = createStore(reducer, composeWithDevTools(middleware));

export default store;
