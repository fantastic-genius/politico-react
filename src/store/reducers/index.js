import { combineReducers } from 'redux';
import auth from '@reducers/Auth';

const reducers = combineReducers({
    auth,
});

export default reducers;
