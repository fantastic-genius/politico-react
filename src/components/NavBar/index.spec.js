import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux';
import store from '@src/store';
import NavBar from './index';

describe('<NavBar />', () => {
  describe('Rendering', () => {
    it('should render NavBar correctly', () => {
        const component = shallow(
          <Provider store={store}>
            <Router>
              <NavBar isAuthenticated={false} />
            </Router>
          </Provider>
        );
        expect(component).toMatchSnapshot();
    });
  })

  describe('Rendering of NavBar and children', () => {
    it('should render NavBar correctly in debug mode', () => {
        const component = mount(
          <Provider store={store}>
            <Router>
              <NavBar isAuthenticated={false} />
            </Router>
          </Provider>
        );
        expect(component).toMatchSnapshot();
        expect(component.find('button')).toHaveLength(2)
    });
    it('should render NavBar correctly in debug mode', () => {
      const component = mount(
        <Provider store={store}>
          <Router>
            <NavBar isAuthenticated={true} />
          </Router>
        </Provider>
      );
      expect(component).toMatchSnapshot();
      expect(component.find('button')).toHaveLength(2)
  });
  })

})
