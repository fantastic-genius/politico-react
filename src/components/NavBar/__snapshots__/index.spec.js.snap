// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Rendering of NavBar and children should render NavBar correctly in debug mode 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NavBar>
      <div>
        <header>
          <div
            className="app-name"
          >
            <Link
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <h2>
                  POLITICO
                </h2>
              </a>
            </Link>
          </div>
          <div
            className="sign-btn"
          >
            <span
              className="home-btn"
            >
              <Link
                to="/signup"
              >
                <a
                  href="/signup"
                  onClick={[Function]}
                >
                  <Button
                    className="btn"
                    value="Sign Up"
                  >
                    <button
                      className="btn"
                    >
                      Sign Up
                    </button>
                  </Button>
                </a>
              </Link>
              <Link
                to="login"
              >
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  <Button
                    className="btn"
                    value="Login"
                  >
                    <button
                      className="btn"
                    >
                      Login
                    </button>
                  </Button>
                </a>
              </Link>
            </span>
          </div>
        </header>
      </div>
    </NavBar>
  </Router>
</BrowserRouter>
`;

exports[`<NavBar /> Rendering should render NavBar correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NavBar />
</Router>
`;
